src/main/java/net/minecraftforge/items/SlotItemHandler.java

```diff
@@ -126,7 +126,8 @@ public int getItemStackLimit(@Nonnull ItemStack stack)
     @Override
     public boolean canTakeStack(EntityPlayer playerIn)
     {
-        return !this.getItemHandler().extractItem(index, 1, true).isEmpty();
+        boolean result = net.minecraftforge.event.ForgeEventFactory.onCanTakeItem(playerIn, this.getStack());
+        return result && !this.getItemHandler().extractItem(index, 1, true).isEmpty();
```

Comment:
Would it be useful for the event to have a "vanilla result" it can reference?

Reply:
Passing the result of `!this.getItemHandler().extractItem(index, 1, true).isEmpty()` to the event, so that modders can see whether players could take the item in vanilla.

Reply:
I see. That could be useful. Is there already a convention used in other events for what this method should be called?

Reply:
Also, the `canTakeStack()` method in the vanilla `Slot` class always returns true. Should we just pass true into the event? or should we add the check manually.

Reply:
Add the check. That way if vanilla changes something down the line, we don't have to.

Reply:
Good point. Now for the name, I was thinking `getAllowed()` or something along those lines.

Reply:
I implemented the check in `SlotItemHandler`, but I was unable to in `Slot` because `canTakeStack()` doesn't have the `ItemStack` as a parameter, and adding that would require editing lots of minecraft files and I thought it would be best to avoid that.